packages/loopring_v3/migrations/2_deploy_libraries.js

```diff
@@ -0,0 +1,29 @@
+// Deploy all libraries that are shared by/linked from other contracts.
+// The purpose of deploying these libraries stand-alone is to reduce gas
+// usage in deploying other contracts.
```

Comment:
Because these libraries only contain `internal` functions, I would think the contracts using them will still include the code inside the contract itself instead of the code in library so I'm not sure this works, but I never tried it.

Reply:
I think you are right. I found this:

> Embedded Library: If a smart contract is consuming a library which have only internal functions than EVM simply embeds library into the contract. Instead of using delegate call to call a function, it simply uses JUMP statement(normal method call). There is no need to separately deploy library in this scenario.